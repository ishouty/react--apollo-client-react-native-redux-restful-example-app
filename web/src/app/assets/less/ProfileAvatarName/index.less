@import "../Colors/index";

.profile-container {
  display: flex;
  align-items: center;
  margin-top: 10px;

  .profile-name {
    display: inline-block;
    top: -12px;

    .name {
      top: 5px;
      width: 130px;
      overflow: hidden;
      text-overflow: ellipsis;
      display: inline-block;
      white-space: nowrap;
    }

    .name-nav {
      .name;
      color: @color-gray;
    }
  }

  .work-position {
    font-size: 11px;
    position: relative;
    font-weight: bold;
    width: 300px;
    overflow: hidden;
    text-overflow: ellipsis;
    display: inline-block;
    white-space: nowrap;
    color: @color-darker-gray-text;
    height: 15px;

  }

}