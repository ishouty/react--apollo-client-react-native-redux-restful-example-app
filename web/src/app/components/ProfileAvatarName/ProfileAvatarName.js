import '../../assets/less/ProfileAvatarName/index.less'
import React, { Component } from 'react'
import ProfileAvatar from './../ProfileAvatar/ProfileAvatar'
import PropTypes from 'prop-types'

class ProfileAvatarName extends Component {
  render() {
    const { profile, size, nav } = this.props

    return profile.picture ? (
      <div className="inline">
        <div className={`profile-container`}>
          <ProfileAvatar
            profile={{ profileImage: profile.picture.medium }}
            size={size}
          />

          <div className="profile-name">
            <div className={nav ? 'name-nav' : 'name'}>
              {profile.name.first} {nav ? '' : profile.name.last}
            </div>
          </div>
        </div>
      </div>
    ) : null
  }
}

ProfileAvatarName.propTypes = {
  profile: PropTypes.object.isRequired,
  nav: PropTypes.string,
  size: PropTypes.string
}

export default ProfileAvatarName
