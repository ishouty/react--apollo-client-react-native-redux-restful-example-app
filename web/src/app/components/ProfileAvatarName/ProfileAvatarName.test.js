import React, {Component} from "react";
import {shallow, mount} from "enzyme"
import toJson from "enzyme-to-json"
import ProfileAvatarName from "./ProfileAvatarName"


describe('ProfileAvatarName', ()=>{

    test('Component should render match snap shot', () => {

        const component = shallow(<ProfileAvatarName profile={ { name: { first: 'mock name' }, picture: { medium: 'thumbnail.png'} }} size="large"/>)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })

    test('Component should render component ProfileAvatar', ()=> {

        const component = shallow(<ProfileAvatarName profile={ { name: { first: 'mock name' }, picture: { medium: 'thumbnail.png'} }} size="large"/>)
        expect(component.find('ProfileAvatar').html()).toEqual('<div class=\"inline\"><div class=\"profile-avatar-large\" style=\"background-image:url(thumbnail.png)\"></div></div>')
        //could not get thiclears to mount and render the child element but was able to render the html mark up using shallow.
    })


})


