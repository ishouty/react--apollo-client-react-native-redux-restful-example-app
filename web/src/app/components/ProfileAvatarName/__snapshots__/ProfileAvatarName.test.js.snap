// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAvatarName Component should render match snap shot 1`] = `
<div
  className="inline"
>
  <div
    className="profile-container"
  >
    <ProfileAvatar
      profile={
        Object {
          "profileImage": "thumbnail.png",
        }
      }
      size="large"
    />
    <div
      className="profile-name"
    >
      <div
        className="name"
      >
        mock name
         
      </div>
    </div>
  </div>
</div>
`;
